File: codeforces/Vacations.cpp
#include <math.h>

#include <iostream>
#include <vector>

typedef int i16;
typedef long int i32;
typedef long long int i64;

using namespace std;

i64 solve(i64 counter, i64 index, i64 n, i64 arr[], bool free_for_contest, bool free_for_gym) {
    if (index == n) return counter;

    switch (arr[index]) {
        case (0):
            return solve(counter + 1, index + 1, n, arr, true, true);
        case (1):
            if (free_for_contest)
                return solve(counter, index + 1, n, arr, false, true);
            else
                return solve(counter + 1, index + 1, n, arr, true, true);
        case (2):
            if (free_for_gym)
                return solve(counter, index + 1, n, arr, true, false);
            else
                return solve(counter + 1, index + 1, n, arr, true, true);
        default:
            if (free_for_contest && !free_for_gym)
                return solve(counter, index + 1, n, arr, false, true);
            else if (free_for_gym && !free_for_contest)
                return solve(counter, index + 1, n, arr, true, false);
            else
                return min(
                    solve(counter, index + 1, n, arr, false, true),
                    solve(counter, index + 1, n, arr, true, false));
    }
}

int main() {
    i64 n;
    cin >> n;

    i64 arr[n];
    for (i64 i = 0; i < n; i++) {
        cin >> arr[i];
    }

    cout << solve(0, 0, n, arr, true, true) << endl;
}

// tle

File: codeforces/Vacations.py
import sys

sys.setrecursionlimit(10**6)


def solve(counter, index, n, arr, free_for_contest, free_for_gym):
    if index == n:
        return counter

    if arr[index] == 0:
        return solve(counter + 1, index + 1, n, arr, True, True)
    elif arr[index] == 1:
        if free_for_contest:
            return solve(counter, index + 1, n, arr, False, True)
        else:
            return solve(counter + 1, index + 1, n, arr, True, True)
    elif arr[index] == 2:
        if free_for_gym:
            return solve(counter, index + 1, n, arr, True, False)
        else:
            return solve(counter + 1, index + 1, n, arr, True, True)
    else:
        if free_for_contest and not free_for_gym:
            return solve(counter, index + 1, n, arr, False, True)
        elif free_for_gym and not free_for_contest:
            return solve(counter, index + 1, n, arr, True, False)
        else:
            return min(
                solve(counter, index + 1, n, arr, False, True),
                solve(counter, index + 1, n, arr, True, False),
            )


def main():
    n = int(input())
    arr = list(map(int, input().split()))

    print(solve(0, 0, n, arr, True, True))


if __name__ == "__main__":
    main()

# tle
